### PR TITLE
SnmpQueryMock: fix OID prefix matching and output format

### PR DESCRIPTION
**Pull request by shward**
Pulled out of #19620 at @murrant's suggestion so the SnmpQueryMock fixes can be reviewed on their own.

Four bugs in the test mock that made it diverge from real net-snmp:

**Prefix matching skipped the dot boundary.** A walk on `1.3.6.1.2.1.2` would also pull in keys under `1.3.6.1.2.1.20`, `1.3.6.1.2.1.21`, and so on - anything sharing the numeric prefix. Real net-snmp terminates on subtree boundary. Tightened to `\$key === \$num_oid || Str::startsWith(\$key, \$num_oid . '.')`.

**OID index suffix was dropped from walk output.** `outputLine()` only knew about the base OID, not the actual row key, so every row in a multi-row walk came back keyed identically. Now passes the row key through and appends the index suffix.

**Lines weren't newline-terminated.** Real `snmpget` and `snmpwalk` end every line with `\n`; the mock didn't, which meant multi-row numeric walks came back as one concatenated unparseable string. Added the trailing newline on every line.

**Numeric output didn't match real net-snmp.** `-On` prints every OID with a leading dot, and `snmpget` on a missing OID returns `No Such Instance currently exists at this OID` instead of an empty value. Both fixed. When `SNMPSIM` is set, the regression test runs the same queries through `NetSnmpQuery` against snmpsim and asserts the mock's raw output is byte-identical.

Most existing tests don't hit these edge cases (which is why they passed before), but anything that walks a subtree whose numeric prefix overlaps a neighbor would silently get extra rows. Worth landing on its own so the diff is reviewable without the noise of an unrelated feature.

No test data is changing in this PR - just the mock behavior.

### CODE REVIEW

**Comment by murrant** [comment]
SnmpQueryMock was only intended to be a partial solution.  As long as this doesn't break anything, seems ok.

However, the CI doesn't use SnmpQueryMock so that won't be of any indication.

File: tests/Mocks/SnmpQueryMock.php [hunk 1]
```diff
@@ -283,24 +283,25 @@ private function getSnmprec(string $community): array
         throw new Exception("SNMPREC: community $community not cached");
     }
 
-    private function outputLine(string $oid, string $num_oid, string $type, string $data): string
+    private function outputLine(string $oid, string $num_oid, string $key, string $type, string $data): string
     {
-        $oid = new Oid($oid);
+        $oidObj = new Oid($oid);
```
Comment (murrant) [inline tests/Mocks/SnmpQueryMock.php:289]: why the oidObj change?

**Comment by murrant** [comment]
It would be really great if you added test that failed before and now succeeds.

**Comment by laf** [comment]
@shward Any update to this?

**Comment by shward** [comment]
@laf I was on vacation and am just getting back to this. Working through #19621 and #19620 feedback now. Will pick up #19651 next, adding the regression test murrant asked for.

**Comment by shward** [comment]
Added a regression test in 147d672 (`tests/Unit/Mocks/SnmpQueryMockTest.php`) with a tiny 5-OID fixture. Covers all three bugs: prefix-overlap matching, dropped OID suffix in walk output, and missing newline termination. Each assertion fails on the pre-fix code and passes after.

File: tests/Mocks/SnmpQueryMock.php (hunk 1 above)
Comment (shward) [reply tests/Mocks/SnmpQueryMock.php:289]: The return now interpolates `$oid$indexSuffix` together, so `$oid` needs to be a string (the original parameter or its translated form), not the parsed `Oid` object. The old code worked by relying on `Oid::__toString`, but doing that mid-concatenation is brittle. Splitting the name makes the two values explicit: the string `$oid` we interpolate at the end, and the parsed `$oidObj` we query for MIB and isNumeric.

File: tests/Unit/Mocks/SnmpQueryMockTest.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+use App\Models\Device;
+use DeviceCache;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use LibreNMS\Tests\DBTestCase;
+use LibreNMS\Tests\Mocks\SnmpQueryMock;
+
+final class SnmpQueryMockTest extends DBTestCase
+{
+    use DatabaseTransactions;
```
Comment (murrant) [inline tests/Unit/Mocks/SnmpQueryMockTest.php]: At first glance, i don't think this test should use the database.

**Comment by murrant** [comment]
TODO, run tests against snmpsim to verify correct target functionality.

**Comment by shward** [comment]
Thanks @murrant, addressed both.

DB dependency: `SnmpQueryMockTest` no longer touches the database. It extends `TestCase` (not `DBTestCase`), drops `DatabaseTransactions`, and fakes the primary device with `DeviceCache::fake()` and an unsaved `Device`, the same pattern as `OpenTSDBStoreTest`. The mock only needs a community string to select its snmprec fixture.

snmpsim TODO: I served the regression fixture with snmpsim and walked it with real net-snmp. The mock output now matches. A scoped walk of `...2.2.1.2` returns only `.2.1/.2.2/.2.3` and excludes the `.20.1/.21.1` siblings (a parent walk confirms those siblings exist in the fixture). Each row carries its own `.1/.2/.3` suffix and ends in `\n` (checked with `od -c`).

Rebased onto current master.

**Comment by murrant** [comment]
I modified your test to run against NetSnmpQuery as well (backed by snmpsim) and your output does not match real net-snmp query output.

**Comment by shward** [comment]
Sorry for all the pedantry in this thread, especially since I then went and missed the actual byte diff :) You were right. Two differences:

- `-On` prints every numeric OID with a leading dot. The mock didn't.
- `snmpget` on a missing OID prints `<oid> = No Such Instance currently exists at this OID` with exit 0, where the mock printed an empty value, so `SnmpResponse::isValid()` disagreed between mock and real.

Both fixed. The test now asserts the match instead of me claiming it: when `SNMPSIM` is set, the same walk/get/getnext run through `NetSnmpQuery` and the raw outputs must be byte-identical. Named output is still approximate, since `-OX` bracket indexes and quoting depend on MIB type info the mock doesn't have.

File: tests/Mocks/SnmpQueryMock.php (hunk 1 above)
Comment (murrant) [reply tests/Mocks/SnmpQueryMock.php:289]: still seems like a change that does nothing.

**Review by murrant — APPROVED** [review]
